client/app/telecom/pack/xdsl/modem/firmware/pack-xdsl-modem-firmware.controller.js

```diff
@@ -0,0 +1,71 @@
+angular.module('managerApp').controller('XdslModemFirmwareCtrl', function ($stateParams, $translate, $q, OvhApiXdsl, TucToast, TucPackXdslModemMediator) {
+  const self = this;
+
+  this.mediator = TucPackXdslModemMediator;
+
+  this.undo = function () {
+    self.firmwareTmp = self.firmware;
+  };
+
+  self.getFirmware = function () {
+    return OvhApiXdsl.Modem().Firmware().v6().get({
+      xdslId: $stateParams.serviceName,
+    }).$promise.then((result) => {
+      self.firmware = Object.values(JSON.parse(JSON.stringify(result))).join('');
```

Comment:
What is the reason of doing `JSON.parse(JSON.stringify…`? I'm curious about it.

Reply:
Because the result of the call Api is a string which is return like this "Resource {0: "1", 1: "7", 2: ".", 3: "1", 4: ".", 5: "7", 6: "9", 7: "6", 8: "0", 9: ".", 10: "0", 11: "1", $promise: Promise, $resolved: true}". After many tries, I didn't find other methods than this. If you have an idea, I take it.

Reply:
In that case I encourage you to update the `$resource`.

Here you can find and example which is quite similar
https://github.com/ovh-ux/ovh-api-services/blob/b558819f9bf5b82dfee679cce2928579f023567f/src/sms/receivers/sms-receivers.v6.service.js#L41-L46

Hope you find this useful!

Reply:
Thanks for your answer.